In [32]:
import pandas as pd

data = pd.read_csv("dataset.csv")
data.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [6]:
print(data["total_items"].max(), data["total_items"].mean())

2000 8.7872


### Question 1

a. The biggest issue with calculating the average order value is that the number of shoes in each order can vary drastically and affect our AOV. We see that there was at least 1 customer who ordered 2000 shoes in 1 order and that the average number of shoes sold per order is 8.8. This means that our AOV of \\$3145 roughly indicates the cost of 9 pairs of shoes which would mean that the average pair of shoes would cost around \\$349. This is much better than the AOV, but I feel like it could be better as common sense and skimming through the data would tell you that \$349 is still a bit too high.

b. The metric that I would report for this dataset is average cost of 1 pair of shoes across all 100 shopify shoe stores after removing any outliers. This metric will take into account orders with a high amount of shoes as well as specialty stores that only sell really expensive shoes to give an accurate understanding of what the average consumer is spending on 1 pair of shoes. 

In [33]:
data["cost_per_item"] = data["order_amount"]/data["total_items"]
average_cost = data["cost_per_item"].sort_values(ascending=False)

# We can safely delete duplicates because we know that each shop only carries 1 type of shoes
average_cost.drop_duplicates(inplace = True)
average_cost

3085    25725.0
1367      352.0
4116      201.0
4458      196.0
4592      195.0
2494      193.0
3689      190.0
4586      187.0
384       184.0
970       181.0
2222      178.0
4171      177.0
2416      176.0
793       173.0
3849      172.0
172       171.0
4078      169.0
4395      168.0
3818      166.0
1128      165.0
4459      164.0
1554      163.0
1368      162.0
28        161.0
2173      160.0
2078      158.0
2753      156.0
2129      155.0
3236      154.0
3945      153.0
1701      149.0
3185      148.0
1033      147.0
2663      146.0
955       145.0
4688      144.0
2658      142.0
886       140.0
363       138.0
128       136.0
2384      134.0
3868      133.0
2486      132.0
1773      131.0
4161      130.0
3463      129.0
2189      128.0
1687      127.0
2489      122.0
4577      118.0
1389      117.0
3514      116.0
2385      114.0
33        112.0
1566      111.0
410       101.0
4684       94.0
158        90.0
Name: cost_per_item, dtype: float64

In [34]:
# Since this is a simple dataset, we can just see and remove the single outlier.
average_cost.drop([3085], axis = 0, inplace = True)
average_cost.mean()

153.57894736842104

c. The average cost of a pair of shoes across 99 shopify stores is \$153.58

### Question 2

<pre>
a. Query: SELECT COUNT(Orders.ShipperID)
          FROM Orders
          INNER JOIN Shippers ON Orders.ShipperID = Shippers.ShipperID\
          WHERE Shippers.ShipperName = 'Speedy Express';
          
Answer: 54 orders were shipped by Speedy Express in total.

b. Query: SELECT LastName, COUNT(LastName) AS NumOrders
          FROM Orders
          INNER JOIN Employees ON Employees.EmployeeID = Orders.EmployeeID
          GROUP BY LastName
          ORDER BY COUNT(LastName) DESC;

Answer: The last name of the employee with the most orders is Peacock.

c. Query: SELECT ProductName, SUM(Quantity) AS Total
          FROM OrderDetails
          LEFT JOIN Products ON OrderDetails.ProductID = Products.ProductID
          WHERE OrderDetails.OrderID IN (SELECT OrderID FROM Orders Left Join Customers ON Orders.CustomerID = Customers.CustomerID WHERE Country = 'Germany')
          GROUP BY ProductName
          ORDER BY SUM(Quantity) DESC;

Answer: Boston Crab Meat is the product that was ordered the most by customers in Germany.
</pre>